public/js/app/components/SemesterDropdown.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState, memo } from 'react'
+import { seasonStr } from '../util/helpers'
+
+function SemesterDropdown({ chooseSemesterLabel, handleSelectedSemester, semesterList, langIndex }) {
+  const [selectedSemester, setSelectedSemester] = useState()
+
+  const handleChange = event => {
+    handleSelectedSemester(event)
+    setSelectedSemester(event.target.value)
+  }
+
+  return (
+    <div className="select-wrapper">
+      <select
+        className="form-control"
+        data-testid="select-terms"
+        id="term-list"
+        aria-label={chooseSemesterLabel}
+        onChange={handleChange}
+        defaultValue="PLACEHOLDER"
+      >
+        {!selectedSemester && (
+          <option key="no-chosen" defaultValue="PLACEHOLDER" style={{ display: 'none' }}>
+            {chooseSemesterLabel}
+          </option>
+        )}
+
+        {semesterList &&
+          semesterList.map(({ term }) => (
+            <option data-testid="select-option" id={`itemFor-${term}`} key={term} value={term}>
+              {seasonStr(langIndex, term)}
+            </option>
+          ))}
+      </select>
+    </div>
+  )
+}
+
+export default memo(SemesterDropdown)
```

Comment:
I cannot remember from our discussion - did using `memo` actually help with something here? As far as I understand - if you address the issue above and remove the local state, `selectedSemester` will most likely be the cause of rerendering anyhow.
As far as I understand it using `React.memo` is not necessary if we are not hunting to improve performance (which I don't think we are).